core/DCache.cpp

```diff
@@ -212,8 +246,40 @@ namespace olympia {
     }
 
     void DCache::issueDownstreamRead_() {
-        // TODO
         // If there is no ongoing refill request
         // Issue MSHR Refill request to downstream
+        if(ongoing_cache_refill_) {
+            return;
+        }
+
+        // Requests are sent in FIFO order
+        current_refill_mshr_entry_ = mshr_file_.begin();
+
+        // Send BIU request
+        // out_biu_req_.send(cache_line);
+        // Temp fix: call event after fixed time (4 cycles)
+        uev_biu_incoming_refill_.schedule(4);
```

Comment:
What's the fix for?  What are your thoughts on the real design here?

Reply:
> I have decoupled the DCache from the BIU. Instead of using the port connected to the BIU, I would schedule an event to simulate the cache refill request.